plugins/package-managers/bazel/src/main/kotlin/CompositeBazelModuleRegistryService.kt

```diff
@@ -0,0 +1,102 @@
+import org.ossreviewtoolkit.clients.bazelmoduleregistry.LocalBazelModuleRegistryService
+import org.ossreviewtoolkit.clients.bazelmoduleregistry.ModuleMetadata
+import org.ossreviewtoolkit.clients.bazelmoduleregistry.ModuleSourceInfo
+import org.ossreviewtoolkit.clients.bazelmoduleregistry.RemoteBazelModuleRegistryService
+
+/**
+ * A composite Bazel module registry service that aggregates multiple [BazelModuleRegistryService] instances and
+ * delegates the package lookup to the corresponding registry. The first registry that contains the package is used.
+ * [packagesPerRegistry] is a map of [BazelModuleRegistryService] instances to the list of packages they are
+ * responsible for.
+ */
+class CompositeBazelModuleRegistryService(
+    private val packagesPerRegistry: Map<BazelModuleRegistryService, List<String>>
+) : BazelModuleRegistryService {
+    companion object {
+        // A regular expression to extract the server and package parts of a registry URL.
+        internal val URL_REGEX = "^(?<server>.*/)modules/(?<package>[^/]+)/[^/]+/source\\.json$".toRegex()
+
+        /**
+         * Create a Composite Bazel Module Registry client instance.
+         */
+        fun create(urls: Set<String>, projectDir: File): CompositeBazelModuleRegistryService {
+            val unknownKey = "UNKNOWN"
+
+            val packagesPerServer = urls.filter { it.endsWith("source.json") }.map {
+                val groups = URL_REGEX.matchEntire(it)?.groups
+
+                val key = groups?.get("server")?.value ?: unknownKey
+                val value = groups?.get("package")?.value ?: it
+
+                key to value
+            }.groupByTo(mutableMapOf(), { it.first }) { it.second }
+
+            packagesPerServer.remove(unknownKey)?.let {
+                logger.warn {
+                    "The following URLs cannot be mapped to a server root:"
+                }
+
+                it.forEach { logger.warn { it } }
+            }
+
+            val packagesPerRegistry = packagesPerServer.mapKeys { entry ->
+                val url = entry.key
+
+                if (url.startsWith("file://")) {
+                    val localRegistryURL = url.replace("%workspace%", projectDir.absolutePath)
+                    logger.info {
+                        "Using local Bazel module registry at '$localRegistryURL'."
+                    }
+
+                    LocalBazelModuleRegistryService(File(localRegistryURL))
+                } else {
+                    RemoteBazelModuleRegistryService.create(url)
+                }
+            }
+
+            return CompositeBazelModuleRegistryService(packagesPerRegistry)
+        }
+    }
+
+    override suspend fun getModuleMetadata(name: String): ModuleMetadata {
+        val registry = packagesPerRegistry.entries.find { name in it.value }
+            ?: throw IllegalArgumentException("No registry found for package '$name'.")
+        // TODO check if other registry entries contain the package
```

Comment:
What "other" registries does this comment refer to ?

Reply:
The rest of the registries if multiple registries have been defined.